executor/write.go

```diff
@@ -929,6 +970,11 @@ func (e *InsertValues) fillRowData(cols []*table.Column, vals []types.Datum, ign
 		row[offset] = v
 		hasValue[offset] = true
 	}
+
+	return e.checkRowData(cols, len(vals), hasValue, row, ignoreErr)
+}
+
+func (e *InsertValues) checkRowData(cols []*table.Column, valLen int, hasValue []bool, row []types.Datum, ignoreErr bool) ([]types.Datum, error) {
 	err := e.initDefaultValues(row, hasValue, ignoreErr)
```

Comment:
why do initDefaultValues again?

Reply:
`fillDefaultValues` just ignore `sql_mode` and use `zero value` or `null` for columns, the result row just a `eval context` so `set a = 1, b = a` can eval correctly. But the actual value for no value column is influenced by `sql_mode` and other conditions, so I simply ask `initDefaultValues` to fill the correct value for no explicit value columns.

Reply:
can we do these check in fillDefaultValues?